cpp/src/arrow/compute/exec.cc

```diff
@@ -473,7 +485,7 @@ bool ExecSpanIterator::Next(ExecSpan* span) {
 namespace {
 
 struct NullGeneralization {
-  enum type { PERHAPS_NULL, ALL_VALID, ALL_NULL };
+  enum type { PERHAPS_NULL = 0, ALL_VALID = 1, ALL_NULL = 2, INVALID = 3 };
```

Comment:
would the name "invalid" ambigious here? Could it be `ALL_NULL` ?

Reply:
Yes, it can be replaced by `ALL_NULL` with `ALL_NULL =3`, so we could keep the original `NullGeneralization::type` of each chunk when doing & operation in `type Get(const ChunkedArray& chunk_array)` .